lib/asn1.ml

```diff
@@ -28,17 +28,42 @@ struct
     }
     [@@deriving ord,eq,show]
 
+    let fix_negative_rsa = function
+      | Ok {n; e} ->
+        if Z.gt n Z.zero then
+          Ok { n; e }
+        else
+          (* A few tools generate keys without leading 0 byte, which gets interpreted
```

Comment:
Comment should say "without a leading 0 bit", not "byte".

Reply:
out of curiousity, which are these "few tools"? I know GnuTLS used to have this behaviour (but IIRC they fixed that some time ago)...

Reply:
Fixed the comment.

We don't have a solid list of affected tools, but it seems a few Microsoft tools (used to?) do this.